pmd-tsql/src/main/java/net/sourceforge/pmd/lang/tsql/cpd/TSqlCpdLexer.java

```diff
@@ -19,4 +23,19 @@ public class TSqlCpdLexer extends AntlrCpdLexer {
     protected Lexer getLexerForSource(CharStream charStream) {
         return new TSqlLexer(new CaseChangingCharStream(charStream, true));
     }
+
+    @Override
+    protected AntlrLexerBehavior getLexerBehavior() {
```

Comment:
This seems to be working... This means, we have now more time to think about the API, if we actually ever need one.

For some reason, the JavaCC implementation is much more complicated (with TokenDocumentBehavior)... But I left this as is.

Reply:
The Javacc implementation is more complicated because it helps with actual Javacc lexing. OTOH Antlr does it lexing alone, and does not use this "token behavior", which is only used by CPD. Hence the difference